archunit/src/main/java/com/tngtech/archunit/library/dependencies/CycleConfiguration.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2014-2020 TNG Technology Consulting GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.tngtech.archunit.library.dependencies;
+
+import com.tngtech.archunit.ArchConfiguration;
+
+final class CycleConfiguration {
+    static final String MAX_NUMBER_OF_CYCLES_TO_DETECT_PROPERTY_NAME = "cycles.maxNumberToDetect";
+    private static final String MAX_NUMBER_OF_CYCLES_TO_DETECT_DEFAULT_VALUE = "100";
```

Comment:
This is a rather low number as a default I think even 1000 will not hit any heap limits on current machines (or build agents).

Reply:
Yes, but what is the info if your Unit test fails and reports 1000 cycles? This was less about "is it doable" as more "is it useful test output". And I think by default to see 100 cycles is way enough to get a feeling for the dimension :wink: 
In particular because of the explosion cycles * edges * violations per edge. Would you really want to read 1000 cycles x 10 edges x 100 dependencies = 1000000 lines of test failure message? Besides that IntelliJ cannot print this anymore as I found out :wink:

Reply:
As below I didn't think of human beeings as consumers of test results in my request.